### PR TITLE
convert some instances of frand_range to Random::next

### PR DESCRIPTION
**Pull request by Goober5000**
Some of the `frand_range` calls were used for integer calculations, so we can use `Random::next` now.

This also reduces the number of C-style casts in the codebase.

### CODE REVIEW

File: code/ship/shipfx.cpp [hunk 1]
```diff
@@ -2474,7 +2474,7 @@ void shipfx_do_lightning_frame( ship * /*shipp*/ )
 	shipp->lightning_stamp = timestamp(stamp);
 
 	// ah, now we can create some lightning bolts
-	count = (int)frand_range(0.0f, (float)count);
+	count = Random::next(0, count);
```
Comment (Baezon) [inline code/ship/shipfx.cpp:2477]: Looks like some of these can be just `Random::next(max)` now? Trivial, but nice to be consistent

File: code/ship/shipfx.cpp [hunk 2]
```diff
@@ -2497,7 +2497,7 @@ void shipfx_do_lightning_frame( ship * /*shipp*/ )
 		binfo.num_strikes = 3;
 		binfo.noise = 0.045f;
 		binfo.life = 375;
-		binfo.delay = (int)frand_range(0.0f, 1600.0f);
+		binfo.delay = Random::next(0, 1600);
```
Comment (Baezon) [inline code/ship/shipfx.cpp:2500]: ditto

File: code/ship/shipfx.cpp [hunk 3]
```diff
@@ -2523,7 +2523,7 @@ void shipfx_do_lightning_frame( ship * /*shipp*/ )
...
```
Comment (Baezon) [inline code/ship/shipfx.cpp:2526]: ditto

File: code/ship/shipfx.cpp [hunk 4]
```diff
@@ -2613,7 +2613,7 @@ void shipfx_do_shockwave_stuff(ship *shipp, shockwave_create_info *sci)
 		sci2.rot_angles.b = frand_range(0.0f, 1.99f*PI);
 		sci2.rot_angles.h = frand_range(0.0f, 1.99f*PI);
 
-		shockwave_create(shipp->objnum, &shockwave_pos, &sci2, SW_SHIP_DEATH, (int)frand_range(0.0f, 350.0f));
+		shockwave_create(shipp->objnum, &shockwave_pos, &sci2, SW_SHIP_DEATH, Random::next(0, 350));
```
Comment (Baezon) [inline code/ship/shipfx.cpp:2616]: ditto

File: code/weapon/beam.cpp [hunk 5]
```diff
@@ -1523,7 +1523,7 @@ void beam_generate_muzzle_particles(beam *b)
 	b->Beam_muzzle_stamp = timestamp(hack_time);
 
 	// randomly generate 10 to 20 particles
-	particle_count = (int)frand_range(0.0f, (float)wip->b_info.beam_particle_count);
+	particle_count = Random::next(0, wip->b_info.beam_particle_count);
```
Comment (Baezon) [inline code/weapon/beam.cpp:1526]: ditto

**Review by Baezon — CHANGES_REQUESTED** [review]

File: code/ship/shipfx.cpp (hunk 1 above)
Comment (Goober5000) [reply code/ship/shipfx.cpp:2477]: Some, but not all.  `Random::next(low, high)` will include both the low and the high value, while `Random::next(modulus)` will include 0 but exclude modulus.  Here, the full count is a valid possibility so the (0, N) call should be used.

Comment (Baezon) [reply code/ship/shipfx.cpp:2477]: Ah, I see. Perhaps `Random::next(max+1)`? It's not a big deal, but it's not super obvious just by looking at it.

Also, on second thought, I believe float casting rounds down, so a frand_range will basically never include its top value anyway.

Comment (Goober5000) [reply code/ship/shipfx.cpp]: Ok, how's this?

Comment (Baezon) [reply code/ship/shipfx.cpp]: Now, I'm starting to have second thoughts lol so I'm just going to approve it before I bikeshed it further and it really isn't a big deal

**Review by Baezon — APPROVED** [review]

**Pull request merged**